File: Group Anagrams - Leetcode 49/Group Anagrams - Leetcode 49.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <algorithm>

using namespace std;

class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> anagramsMap;
        
        for (const string& s : strs) {
            string count = getCount(s);
            anagramsMap[count].push_back(s);
        }
        
        vector<vector<string>> result;
        for (const auto& entry : anagramsMap) {
            result.push_back(entry.second);
        }
        
        return result;
    }
    
private:
    string getCount(const string& s) {
        vector<int> count(26, 0);
        for (char c : s) {
            count[c - 'a']++;
        }
        string countStr;
        for (int i : count) {
            countStr += to_string(i) + '#';
        }
        return countStr;
    }
};

File: Group Anagrams - Leetcode 49/Group Anagrams - Leetcode 49.java
import java.util.*;

class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> anagramsMap = new HashMap<>();
        
        for (String s : strs) {
            int[] count = new int[26];
            for (char c : s.toCharArray()) {
                count[c - 'a']++;
            }
            StringBuilder key = new StringBuilder();
            for (int num : count) {
                key.append('#').append(num);
            }
            String keyStr = key.toString();
            anagramsMap.computeIfAbsent(keyStr, k -> new ArrayList<>()).add(s);
        }
        
        return new ArrayList<>(anagramsMap.values());
    }
}
